src/catalog/mod.rs

```diff
@@ -435,6 +438,58 @@ pub async fn get_first_event(
     Ok(Some(first_event_at))
 }
 
+/// Retrieves the earliest first-event-at from the storage for the specified stream.
+///
+/// This function fetches the object-store format from all the stream.json files for the given stream from the storage,
+/// extracts the `first_event_at` timestamps, and returns the earliest `first_event_at`.
+///
+/// # Arguments
+///
+/// * `stream_name` - The name of the stream for which `first_event_at` is to be retrieved.
+///
+/// # Returns
+///
+/// * `Result<Option<String>, ObjectStorageError>` - Returns `Ok(Some(String))` with the earliest
+///   first event timestamp if found, `Ok(None)` if no timestamps are found, or an `ObjectStorageError`
+///   if an error occurs.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// * The stream metadata cannot be retrieved from the storage.
+///
+/// # Examples
+/// ```ignore
+/// ```rust
+/// let result = get_first_event_from_storage("my_stream").await;
+/// match result {
+///     Ok(Some(first_event)) => println!("first-event-at: {}", first_event),
+///     Ok(None) => println!("first-event-at not found"),
+///     Err(e) => eprintln!("Error retrieving first-event-at from storage: {}", e),
+/// }
+/// ```
```

Comment:
The code comments aren't very helpful, can we restrict to only pointing out working and edge-cases?